### ISSUE TITLE
Unable to capture SSL traffic when NOT connected to the VPN

### ISSUE THREAD

**Issue by albertbori** [ProxymanApp/Proxyman#1743]
## Description

This is a really odd case. All of the help articles and troubleshooting resources I can find focus on how to get Proxyman to work while your VPN _is running_. However, when I disconnect and/or quit my VPN client, Proxyman will no longer correctly proxy any SSL traffic... ever... (I get `CONNECT 999 Internal Error` traffic for the domains in question.)

No amount of restarting Proxyman, restarting the MacBook, or reinstalling Proxyman seems to fix the issue.

The SSL proxying works fantastically well, when connected to the VPN. Unfortunately I have some scenarios that need to be tested outside of the VPN connection and I'm unable to test those scenarios.

Notes: 
- Charles Proxy appears to proxy SSL traffic fine while disconnected from the VPN.
- Physical devices connected to Proxyman over the network proxy SSL traffic fine wile disconnected from the VPN.
- I have attempted to add the VPN domains to the Proxy Bypass list, to no avail.

## Steps to Reproduce

1. Launch Proxyman on an M1 MacBook Pro
2. Launch and connect to VPN via Cisco AnyConnect
3. Launch and test an iOS app via the iOS Simulator
4. See that SSL traffic is successfully proxied in Proxyman
5. Disconnect from the VPN and/or quit the Cisco AnyConnect app
6. Launch and test an iOS app via the iOS Simulator

## Current Behavior

In Proxyman, experience `CONNECT 999 Internal Error` on all SSL traffic. No workarounds.

## Expected Behavior

In Proxyman, see normal/healthy SSL traffic

## Environment

- App version: 4.9.1 49001
- macOS version: macOS Ventura 13.5 22G74
- VPN: Cisco AnyConnect (4.10.03104)
- Device Hardware: M1 MacBook Pro 14" 2021 32GB Ram

**Comment by NghiaTranUIT** [comment]
It's difficult for me to reproduce it because I don't have Cisco AnyConnection VPN.

Just wondering: After the 5th step (`5. Disconnect from the VPN and/or quit the Cisco AnyConnect app`), can you open the System Setting -> Wifi -> Your Wifi -> Details -> Proxies Tab -> Check if the HTTP/HTTPS Proxy is enabled? 🤔 

And if it's enabled, what values (IP, Port) are?

I assume that when the VPN is closed, it changes the HTTP/HTTPS Proxy value, so your traffic from your Mac is always disconnected.

![CleanShot 2023-08-18 at 09 31 27@2x](https://github.com/ProxymanApp/Proxyman/assets/5878421/b3d15fff-a44a-4623-ae3a-30e64fa98135)

**Comment by albertbori** [comment]
Thanks for the quick reply. Here is the screenshot of my proxies view for the active network connection (Ethernet through a thunderbolt docking station). After Step 5, I opened the following settings menu and took a screenshot:
![image](https://github.com/ProxymanApp/Proxyman/assets/1742709/4734bd4c-c986-4ef3-b2a9-72574a112753)

**Comment by NghiaTranUIT** [comment]
@albertbori can you follow this:
1. When you get this 999 error -> Help menu -> Advacend -> Enable Debug Mode.
2. Close Proxyman app and start the app again by using the command line: `/Applications/Proxyman.app/Contents/MacOS/Proxyman`
3. Try to reproduce the issue again, and share with me the error log.

Second question: Do you install the Helper Tools in Setting -> Advance Tab -> Helper Tool? 🤔

**Comment by albertbori** [comment]
> Second question: Do you install the Helper Tools in Setting -> Advance Tab -> Helper Tool? 🤔
I'm pretty sure I installed this (I used the automatic setup). How do I verify that the helper tool is installed?

**Comment by NghiaTranUIT** [comment]
It's in the Setting -> Advance Tab -> Helper Tool

![CleanShot 2023-08-25 at 15 50 02@2x](https://github.com/ProxymanApp/Proxyman/assets/5878421/28e8726b-1b96-4971-b446-b0e5a9152c5a)

**Comment by albertbori** [comment]
>>> Second question: Do you install the Helper Tools in Setting -> Advance Tab -> Helper Tool? 🤔

>> I'm pretty sure I installed this (I used the automatic setup). How do I verify that the helper tool is installed?

> It's in the Setting -> Advance Tab -> Helper Tool

Ok, thank you. I verified that it was installed.

> When you get this 999 error -> Help menu -> Advacend -> Enable Debug Mode.
> 1. Close Proxyman app and start the app again by using the command line: `/Applications/Proxyman.app/Contents/MacOS/Proxyman`
> 2. Try to reproduce the issue again, and share with me the error log.

I attempted to recreate this issue so that I could follow the "Debug Mode" instructions you gave me above, but, to my surprise, Proxyman started capturing SSL traffic while disconnected from the VPN... I tried to reproduce the issue several more times, but Proxyman worked correctly each time. `¯\_(ツ)_/¯` This is extremely odd to me because I've been able to reproduce this issue 100% of the time for the past couple of months. I even did several extensive and deep troubleshooting sessions to attempt to correct the issue in the past.

Since I don't know why this issue suddenly self-resolved, I hesitate to call this issue "fixed". However, I can no longer reproduce the issue at this time, so we can probably close this ticket.

**Comment by albertbori** [comment]
Update: I discovered some potentially helpful info: This issue is isolated to traffic coming from the iOS Simulator. Just to be sure it wasn't the certificate, I did:
1. Certificate > Install Certificate on iOS > Simulators...
2. Install Root Proxyman Certificate to this Mac: ✅ Installed
3. Install and Trust the certificate to running Simulators: ✅ Installed
4. Reboot iOS Simulators: ✅ Rebooted!
![image](https://github.com/ProxymanApp/Proxyman/assets/1742709/4d5b0e9e-5cd6-451a-8e58-7a78cc7fe314)

**Comment by NghiaTranUIT** [comment]
Sorry for the inconvenience, it's all about the AnyAconnect VPN, which is incompatible with Proxyman and Charles Proxy.

Not using VPN + Proxyman at the same time will always fix the issue.

**Comment by albertbori** [comment]
I have good news! I did not find the source of the issue, but I did find a workaround. I was able to "un-break" the SSL proxying feature by toggling my wifi on/off and disconnecting and reconnecting my ethernet cable.

Backstory: I am connected to ethernet via a USB-C hub. Since some services (namely AnyConnect) often don't respect Network Service Order, I disabled my wifi adapter while connected to ethernet. At some point, the SSL proxying feature stops working _only for the iOS simulator_ with that configuration. Not even rebooting would fix it.

By reenabling my wifi, and disconnecting/reconnecting my ethernet, the SSL proxying began working while disconnected from the VPN again.

I'm not sure whose fault this is, but I don't think it's Proxyman's. If I had to guess, it'd be either AnyConnect or Apple's iOS Simulator software.

Either way, we can close this issue as there's no reasonable path nor expectation for getting a permanent resolution for such an edge case, and there's a clear and consistent workaround.

Thanks for your help and support while I was troubleshooting this issue! 🙏
